### ISSUE TITLE
azurerm_cdn_endpoint doesn't notice if origin_path is removed or set to ""

### ISSUE THREAD

**Issue by Supermathie** [terraform-providers/terraform-provider-azurerm#1595]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.azurerm v1.9.0
+ provider.random v1.3.1
+ provider.template v1.0.0

### Affected Resource(s)

* azurerm_cdn_endpoint

### Terraform Configuration Files

```hcl
resource "azurerm_cdn_endpoint" "app" {
  name                = "${terraform.workspace}-${var.label}"
  location            = "${azurerm_resource_group.app.location}"
  resource_group_name = "${azurerm_resource_group.app.name}"
  profile_name        = "${azurerm_cdn_profile.app.name}"

  optimization_type   = "GeneralWebDelivery"
  origin_path         = "/${azurerm_storage_container.assets.name}/"
  #origin_path         = ""
  #origin_path         = "/"
  origin_host_header  = "${azurerm_storage_account.app.name}.blob.core.windows.net"
  origin {
    name = "assets"
    host_name = "${azurerm_storage_account.app.name}.blob.core.windows.net"
  }
  tags {
    Cluster = "${var.label}"
    DC = "${terraform.workspace}"
  }
}
```

### Debug Output

(available on request)

### Expected Behavior

I have an existing endpoint successfully created that looks like this:

```text
module.discuss_cluster.azurerm_cdn_endpoint.app:
  id = /subscriptions/xxxx/resourcegroups/xxxx-dev-michael-discuss/providers/Microsoft.Cdn/profiles/discuss/endpoints/xxxx-dev-michael-discuss
  content_types_to_compress.# = 0
  geo_filter.# = 0
  host_name = xxxx-dev-michael-discuss.azureedge.net
  is_compression_enabled = false
  is_http_allowed = true
  is_https_allowed = true
  location = eastus
  name = xxxx-dev-michael-discuss
  optimization_type = GeneralWebDelivery
  origin.# = 1
  origin.722596397.host_name = xxxxdevmichaeldiscuss.blob.core.windows.net
  origin.722596397.http_port = 80
  origin.722596397.https_port = 443
  origin.722596397.name = assets
  origin_host_header = xxxxdevmichaeldiscuss.blob.core.windows.net
  origin_path = /assets/
  probe_path = 
  profile_name = discuss
  querystring_caching_behaviour = IgnoreQueryString
  resource_group_name = xxxx-dev-michael-discuss
  tags.% = 2
  tags.Cluster = discuss
  tags.DC = xxxx-dev-michael
```

I attempted to introduce a change by commenting `origin_path` and re-running `terraform apply` but terraform didn't detect the change.

### Actual Behavior

"No changes. Infrastructure is up-to-date."

### Steps to Reproduce

1. comment out `origin_path`
1. `terraform apply`

Setting it explicitly to `""` also did not cause a change.

I had to set it to another value (`"/"`) to force the change (though `/` is invalid here so it didn't apply):

```text
Terraform will perform the following actions:

  ~ module.discuss_cluster.azurerm_cdn_endpoint.app
      origin_path: "/assets/" => "/"
```

### References

Ø

**Comment by Supermathie** [comment]
Same for `origin_host_header`

**Comment by tombuildsstuff** [comment]
Fixed via #7164
